google-oauth-client/src/main/java/com/google/api/client/auth/openidconnect/IdTokenVerifier.java

```diff
@@ -141,10 +194,60 @@ public final String getIssuer() {
    * @param idToken ID token
    * @return {@code true} if verified successfully or {@code false} if failed
    */
-  public boolean verify(IdToken idToken) {
-    return (issuers == null || idToken.verifyIssuer(issuers))
-        && (audience == null || idToken.verifyAudience(audience))
-        && idToken.verifyTime(clock.currentTimeMillis(), acceptableTimeSkewSeconds);
+  public boolean verify(IdToken idToken) throws VerificationException {
```

Comment:
Would you improve the Javadoc that explains the newly-added case?

Reply:
I'm surprised the breaking change detector doesn't see adding a checked exception as a breaking change.

It also feels a bit weird to have 2 different mechanisms for indicating an invalid token. In the google-auth-library implementation, we only throw `VerificationException`s (to be able to communicate why it failed). This legacy implementation already is hiding why the token in invalid.

Reply:
Another option, in line with Tomo's suggestion is to add a new `verifySignature` method that throws `VerificationException`. `verify` could call `verifySignature` and catch the exception and return `false` (to not change the interface)